ansible/roles/containerd/tasks/install-flatcar.yaml

```diff
@@ -5,6 +5,41 @@
     src: etc/systemd/system/containerd-flatcar.service
     mode: 0600
 
+- name: Download docker for 3033.3.x flatcar LTS to /opt/docker.tgz
+  when:
+    - ansible_facts['distribution_version'] is version('3033.3.0', '>=', version_type='semver')
+    - ansible_facts['distribution_version'] is version('3033.4.0', '<', version_type='semver')
```

Comment:
Are we sure that none of the `3033.3.0` versions will have a newer version of Containerd?
I think the more important part to check here is the docker version, and only install the new version if its <`20.10.15`, this would be safer and avoid potential downgrades.

You can run probably run `docker version -f json` and store that state into a variable.

Reply:
https://www.flatcar.org/releases#release-3033.3.0 was shipped with `20.10.12` and docker version is not bumped in patch releases (https://www.flatcar.org/releases#release-3033.3.13 has the same docker version).

I initially thought of looking at containerd version (or we can do docker version as you suggested) but, if for some reason, this is run against a different lts release then i dont think we want to run this task (say, to avoid upgrading docker from 19.x.y -> 20.x.y which would be a major version bump). I agree this is highly unlikely but I think we can retain the `distribution_check` and add docker version check as an additional condition.

Reply:
Great point, I think adding both checks sounds great.

Also you can wrap all of these tasks into a `block` with this `when` condition since we really only want to do all of this selectively.